src/components/appointments/ChooseProductStep.js

```diff
@@ -1,77 +1,134 @@
 import {FieldArray, useFormikContext} from 'formik';
 import PropTypes from 'prop-types';
 import React from 'react';
-import {FormattedMessage} from 'react-intl';
+import {FormattedMessage, useIntl} from 'react-intl';
 
 import Button from 'components/Button';
+import ButtonsToolbar from 'components/ButtonsToolbar';
+import Card, {CardTitle} from 'components/Card';
 import FAIcon from 'components/FAIcon';
 import {Toolbar, ToolbarList} from 'components/Toolbar';
+import {SUBMISSION_ALLOWED} from 'components/constants';
 import {getBEMClassName} from 'utils';
 
 import Product from './Product';
 
-const ChooseProductStep = () => {
+const isValidProduct = ({product, amount}) => {
+  if (!product) return false;
+  if (amount <= 0) return false;
+  return true;
+};
+
+// TODO: replace with ZOD validation, see #435
+const isStepValid = data => {
+  const products = data?.products;
+  if (!products || products.length <= 0) {
+    return false;
+  }
+  const allProductsComplete = products.every(isValidProduct);
+  return allProductsComplete;
+};
+
+const ChooseProductStep = ({formName}) => {
+  const intl = useIntl();
   const {values} = useFormikContext();
+
   const products = values?.products || [];
   const numProducts = Math.max(products.length, 1);
   return (
-    <FieldArray name="products">
-      {arrayHelpers => (
-        <div className={getBEMClassName('editgrid')}>
-          <div className={getBEMClassName('editgrid__groups')}>
-            {products.map(({product: productId}, index) => (
-              // blank blocks don't have a product selected yet -> so the index is added
-              // to make the key guaranteed unique
-              <div key={`${productId}-${index}`} className={getBEMClassName('editgrid__group')}>
-                <div className={getBEMClassName('editgrid__group-label')}>
-                  <FormattedMessage
-                    description="Appointments: single product label/header"
-                    defaultMessage="Product {number}/{total}"
-                    values={{number: index + 1, total: numProducts}}
-                  />
-                </div>
+    <Card title={formName} titleComponent="h1" modifiers={['mobile-header-hidden']}>
```

Comment:
TODO: check if this can be moved into the parent component instead, as it applies to all steps.